### ISSUE TITLE
Custom fields with possibility of not only text

### ISSUE THREAD

**Issue by jojo12** [jemproject/JEM-Project#1667]
There's a post which asks for savehtml

**Comment by Heklaterriol** [comment]
I played around (in the backend):

modify the custom field information `event.xml` 
/administrator/components/com_jem/models/forms/event.xml

e.g. for custom field 1:

Change from
```
<field name="custom1" type="text"
	size="20"
	readonly="false"
	class="inputbox"
	label="COM_JEM_EVENT_CUSTOM_FIELD1"
	description="COM_JEM_EVENT_CUSTOM_FIELD2_DESC"
/>
```
to
```
<field name="custom1" type="editor"
	height="200px"
	size="20"
	readonly="false"
	class="inputbox"
	filter="JComponentHelper::filterText"
	label="COM_JEM_EVENT_CUSTOM_FIELD1"
	description="COM_JEM_EVENT_CUSTOM_FIELD1_DESC"
/>
```

This way you get the custom fields with editor and allows everything which is allowed in the "Event Description" textarea. 
The custom fields are limited to 200 characters.

It could result in chaotic content in the custom fields – would be good to allow only basic html or similar.

**Comment by Heklaterriol** [comment]
Just read the post in the forum.
`type="safehtml"` is also possible, and concerning the interface even smoother, because not the whole editor is loaded

**Comment by Heklaterriol** [comment]
@Jojo and @mckillo
do you agree to change the code of th custom fileds to type="safehtml" etc.?
I think it is a good idea

**Comment by Heklaterriol** [comment]
My thoughts about this:
I think it is good to have html code not filteret out, but an editor is too much and an open gate to issues.
The ones who know enough html can use it.

**Comment by jojo12** [comment]
Note: The custom fields are problematic, because their use depends on the administration, how is decided to use each field in a site.
e.g. event custom field 1 was used from stevent for the artist of an event (because of this it's the only field with 200 characters.
The other fields had only 100 characters.

**Comment by Heklaterriol** [comment]
OK I see: field 1 and 2 have varchar(200) the rest varchar(100).

**My suggestion:**

We set every field up to 200 chars and add  `filter="safehtml"`.

This way for existing users all stays the same, except for the ones who try html.
In addition, setting all fields to 200 chars is more consisten (=less confusing).

That means we change in field 3-10:
```
	<field name="customX" type="text"
			   filter="safehtml"
			   size="20"
			   readonly="false"
			   class="inputbox"
			   label="COM_JEM_EVENT_CUSTOM_FIELDX"
			   description="COM_JEM_EVENT_CUSTOM_FIELDX_DESC"
		/>
```
to
```
		<field name="customX" type="text"
			   size="20"
			   readonly="false"
			   class="inputbox"
			   label="COM_JEM_EVENT_CUSTOM_FIELDX"
			   description="COM_JEM_EVENT_CUSTOM_FIELDX_DESC"
		/>
```
and
```
ALTER TABLE `#__jem_events`
CHANGE `custom3` `custom3` VARCHAR(200),
CHANGE `custom4` `custom4` VARCHAR(200),
CHANGE `custom5` `custom5` VARCHAR(200),
CHANGE `custom6` `custom6` VARCHAR(200),
CHANGE `custom7` `custom7` VARCHAR(200),
CHANGE `custom8` `custom8` VARCHAR(200),
CHANGE `custom9` `custom9` VARCHAR(200),
CHANGE `custom10` `custom10` VARCHAR(200); 

```

What do you think?

**Comment by jojo12** [comment]
I'm not sure, but there was (years ago) a discussion why we have to limit the possibilities of custom fields to minimize abuse.
But I don't find this. Perhaps we have to find these reasons and have answers.

**Comment by Heklaterriol** [comment]
I think it is good to have at least limited html allowed (only styling) but none with advanced possibilities like `iframe` etc.

**Comment by mckillo** [comment]
In my opinion, I think that in this case, the admin can do a view override. 
Please, no iframe in JEM. 
I don't see the need to put these fields by default in HTML
The admin can set up these custom fields by CSS (class from custom1 to custom10 for dt and dd). It's enough to custom these fields, and the content must text/plain, as all fields of the event. The admin has a field in HTML --> description of the event (and other in the venue description, if it's shown), even include article into the description.
We don't need to devs any extra code of filters/security content for these fields.

**Comment by mckillo** [comment]
I agree that the current custom fields are limited, but changing one of the existing custom1-custom10 fields to allow HTML is not the right solution. These fields are generic text fields and are used differently on existing sites, so selecting one fixed field for safe HTML would be arbitrary and could also introduce security and output escaping issues.

The custom fields area is planned to be reviewed and redesigned for JEM 5.1.0. For that reason, we prefer not to make a partial change now. The redesign should handle field types and filtering in a more consistent and configurable way.

**Comment by mckillo** [comment]
Now, custom fields can be changed from the backend.
Go to Settings → Custom Fields.

<img width="2147" height="999" alt="Image" src="https://github.com/user-attachments/assets/45f5ae2e-c269-4519-a842-25a5c5e711e5" />

See more info: https://github.com/jemproject/JEM-Project/issues/2207
